### ISSUE TITLE
Historical networth

### ISSUE THREAD

**Issue by andreva7** [MoneyTools/MyMoney.Net#130]
Historical net worth does not handle the stock price missing on certain dates.
If it does not find the price for the day selected, it picks 0 from the price instead of searching backwards for the most recent price prior t the day in question.

A small feature addition would be to add a yellow flag if the historical data is missing and myMoney is having to use this heuristic for evaluate the value.

**Comment by clovett** [comment]
If you still have this problem take a look at the data you have in the XML files in your StockQuotes folder.  If the dates are also missing there then you probably need to run "Online/Update Securities" and wait for it to completely finish.  The output window now shows more progress so you know when it is still running if it ends with this:

> TwelveData stock quote service needs to sleep for seconds 61

Instead of this:
>TwelveData download complete

This is also an initial "TwelveData download complete" which you should ignore, so you should see 2 "TwelveData download complete" prompts to know that it has truly finished.  For example, right now I see this:

```
Fetching Stock Quotes
TwelveData download complete
Quote received for VOD, closed on 8/1/1997 at $11.65
Quote received for ABT, closed on 8/1/1997 at $14.29
Quote received for LSBDX, closed on 4/11/2011 at $14.71
Quote received for ORNAX, closed on 8/1/1997 at $11.42
Quote received for OIBAX, closed on 8/1/1997 at $5.48
Quote received for NSRGY, closed on 8/1/1997 at $12.65
Quote received for NTDOY, closed on 8/1/1997 at $2.43
Quote received for FLEX, closed on 8/1/1997 at $3.75

TwelveData stock quote service needs to sleep for seconds 61
Quote received for CAH, closed on 8/1/1997 at $19.33
Quote received for TESIX, closed on 8/1/1997 at $21.91
Quote received for DHI, closed on 8/1/1997 at $3.51
```

Which means it is still running and is filling in some old dates back to 1997. 

If you have a paid subscription it will all run much faster because the paid subscription has no daily limits.

**Comment by andreva7** [comment]
OK

**Comment by andreva7** [comment]
The stock price tables do not have entries for weekends!
The bar chart today showed a very low number for 2023/12/31. Turns out MyMoney thoughts MSFT was 29$ a year ago.
I looked at the stock quotes and 12/31 last year was on a weekend

**Comment by clovett** [comment]
Correct, the stock price XML files only have data for days that the stock market is open.  I don't know where it could have gotten $29 from but a missing weekend is not the cause of the problem.  I create a price by date index in memory from these XML files and the lookup from that index walks back up to 30 days looking for a stock price.  If none is found then it searches nearby transactions for a trade containing a stock price and uses that.

Perhaps we should just try deleting all your XML files in the StockQuotes folder and re-download them from TwelveData then see what happens.

**Comment by andreva7** [comment]
I can no longer repro the problem. That's a little scary to me since it reproed consistently before, but oh well. Nothing to do right now.
Thanks.

-Andre

**Comment by clovett** [comment]
Ok, good to know, I'll close it for now, and in the meantime I'll do some more testing.  Let me know if you see anything else wonky.
